service/history/replication/task_processor_test.go

```diff
@@ -481,20 +532,69 @@ func (s *taskProcessorSuite) TestTriggerDataInconsistencyScan_Success() {
 	s.NoError(err)
 }
 
-type fakeTaskFetcher struct {
-	sourceCluster string
-	requestChan   chan<- *request
-	rateLimiter   *quotas.DynamicRateLimiter
+func (s *taskProcessorSuite) TestCleanupReplicationTaskLoop() {
+	wantResp := &persistence.RangeCompleteReplicationTaskResponse{
+		TasksCompleted: 15, // if this number is different than page size the loop breaks
```

Comment:
we should test the range delete loop. Add another mock which expects 50 tasks to be deleted.

Reply:
missed to cover that case. done